packages/theme/src/cli/utilities/theme-environment/hot-reload/server.ts

```diff
@@ -187,12 +185,19 @@ export function getHotReloadHandler(theme: Theme, ctx: DevServerContext) {
       }
 
       const replaceTemplates: {[key: string]: string} = {}
-
       const inMemoryTemplateFiles = ctx.localThemeFileSystem.unsyncedFileKeys
-      const sectionTemplate =
-        inMemoryTemplateFiles.has(sectionKey) && ctx.localThemeFileSystem.files.get(sectionKey)?.value
 
-      if (sectionTemplate) replaceTemplates[sectionKey] = sectionTemplate
+      if (inMemoryTemplateFiles.has(sectionKey)) {
+        const sectionTemplate = ctx.localThemeFileSystem.files.get(sectionKey)?.value
+        if (!sectionTemplate) {
+          // If the section template is not found, it means that the section has been removed.
+          // The remote version might not yet be synced so, instead of rendering it remotely,
+          // which should return an empty section, we directly return the same thing here.
+          return ''
+        }
+
+        replaceTemplates[sectionKey] = sectionTemplate
+      }
```

Comment:
@jamesmengo This is what we were missing during our pairing to fix the empty files. The server was falling back to the remote renderer because `value` was empty, and we were not passing it in `replaceTemplates`.